libs/stats/odc/stats/_cli_save_tasks.py

```diff
@@ -169,26 +169,17 @@ def gqa_predicate(ds):
 def _parse_products(dc, products, temporal_range):
         
     query = {"product": products}
-    query.update(
-            temporal_range.dc_query(pad=0.6)
-        ) 
-    dss = ordered_dss(
-        dc, key=lambda ds: (ds.center_time, ds.metadata.region_code), **query
-    )
+    query.update(temporal_range.dc_query(pad=0.6)) 
+    dss = ordered_dss(dc, key=lambda ds: (ds.center_time, ds.metadata.region_code), **query)
     paired_dss = groupby(dss, key=lambda ds: (ds.center_time, ds.metadata.region_code))
-    paired_dss = [list(t[1]) for t in paired_dss]
-    bad_dss = [x for x in paired_dss if len(x) != 2]
-    paired_dss = [x for x in paired_dss if len(x) == 2]
-    
-    for x in groupby(bad_dss, key=lambda ds: ds[0].type.name):
-        name, _iter = x
-        print(f"Warning: product {name} has {len(list(_iter))} unpaired datasets")
-    
+    paired_dss = (tuple(t[1]) for t in paired_dss)
```

Comment:
using unpacking syntax here is preferable to indexing, something like
```python
paired_dss = (tuple(group) for _, group in in paired_dss)
```

in fact you can combine lines above and below as

```python
paired_dss = (tuple(group) for _, group in in paired_dss if len(group) == expected_group_size)
```